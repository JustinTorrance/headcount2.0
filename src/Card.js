import React from 'react';
import './styles/Card.css';
import PropTypes from 'prop-types';


const Card = ({location, stats}) => {
  const year = Object.keys(stats).map(year => (year))
  const value = Object.values(stats).map(value => (value))
  const yearlyValue = year.map((yearlyValue, i) => {
    return <li className={ value[i] > .5 ? 'green' : 'red'} >{year[i]}: {value[i]}</li>
  })

  return(
    <article className='school-card'>
      <h3>{location}</h3>
        <ul>
          {yearlyValue}
        </ul>
    </article>
  );
}

Card.propTypes = {
  stats: PropTypes.object.isRequired,
  location: PropTypes.string.isRequired
}

export default Card;