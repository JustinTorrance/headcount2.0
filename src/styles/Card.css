.school-card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 15px;
  border: black solid 1px;
  padding: 15px 10px;
  -webkit-box-shadow: 3px 3px 5px 6px #ccc;  /* Safari 3-4, iOS 4.0.2 - 4.2, Android 2.3+ */
  -moz-box-shadow:    3px 3px 5px 6px #ccc;  /* Firefox 3.5 - 3.6 */
  box-shadow:         3px 3px 5px 6px #ccc;  /* Opera 10.5, IE 9, Firefox 4+, Chrome 6+, iOS 5 */
}

.school-card h3 {
  text-align: center;
  margin-bottom: 10px;
}

.school-card li {
  margin-left: 33%;
  list-style-type: none;
}

.red {
  color: red;
}

.green {
  color: green;
}