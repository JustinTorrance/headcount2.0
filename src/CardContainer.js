import React from 'react';
import Card from './Card.js';
import './styles/CardContainer.css';
import PropTypes from 'prop-types';


const CardContainer = ({ findAllMatches, searchedLocation }) => {

  const matches = findAllMatches(searchedLocation)
  const schoolCard = matches.map(match => ( <Card {...match} /> ));
  return(
    <section className='card-container'>
      { schoolCard }
    </section>
  )
}

CardContainer.propTypes = {
  findAllMatches: PropTypes.func.isRequired
}

export default CardContainer;

