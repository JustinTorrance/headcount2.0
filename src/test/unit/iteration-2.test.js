import React from 'react';
import Card from '../../Card.js';
import CardContainer from '../../CardContainer.js';
import {shallow, mount} from 'enzyme';
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';


describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <Card
        location='colorado'
        stats = { {2004: 1} }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
})

describe('CardContainer', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const repository = new DistrictRepository(kinderData);
    const wrapper = mount(<CardContainer findAllMatches={ repository.findAllMatches } />)
    expect(wrapper).toMatchSnapshot();
  })
})
