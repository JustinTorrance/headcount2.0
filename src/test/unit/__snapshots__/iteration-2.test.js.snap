// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    location="colorado"
    stats={
        Object {
            "2004": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          colorado
</h3>,
        <ul>
          <li
                    className="green"
          >
                    2004
                    : 
                    1
          </li>
</ul>,
      ],
      "className": "school-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "colorado",
        },
        "ref": null,
        "rendered": "colorado",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="green"
>
              2004
              : 
              1
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                1,
              ],
              "className": "green",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              1,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            colorado
</h3>,
          <ul>
            <li
                        className="green"
            >
                        2004
                        : 
                        1
            </li>
</ul>,
        ],
        "className": "school-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "colorado",
          },
          "ref": null,
          "rendered": "colorado",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="green"
>
                2004
                : 
                1
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  1,
                ],
                "className": "green",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                1,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
